researchgraph/llmnode/llmnode.py

```diff
@@ -17,7 +18,7 @@
 logger = logging.getLogger("researchgraph")
 
 
-class State(TypedDict):
+class State(BaseModel):
```

Comment:
I think it's fine to keep using TypedDict for LangGraph's State!
It looks like TypedDict is used in [the official documentation](https://langchain-ai.github.io/langgraph/tutorials/introduction/#part-3-adding-memory-to-the-chatbot), rather than pydantic.

Reply:
The same applies to other parts.